Dashboard/app/js/lib/components/devices/AssignmentsEditView/index.jsx

```diff
@@ -0,0 +1,210 @@
+/* eslint-disable import/no-unresolved */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import FolderSurveySelectorView from 'akvo-flow/components/selectors/FolderSurveySelector';
+import FormSelectorView from 'akvo-flow/components/selectors/FormSelector';
+import DeviceGroupSelectorView from 'akvo-flow/components/selectors/DeviceSelector';
+import AssignmentDetails from './AssignmentDetails';
+
+export default class AssignmentsEditView extends React.Component {
+  state = {
+    assignmentName: this.props.inputValues.assignmentName,
+    startDate: this.props.inputValues.startDate,
+    expireDate: this.props.inputValues.toDate,
+    nameValidationMsg: '',
+    expireDateMsg: '',
+    startDateMsg: '',
+  }
+
+  // lifecycle methods
+  componentDidMount() {
+    this.validateAssignmentName(this.props.inputValues.assignmentName);
+    this.validateDate('startDateMsg', this.props.inputValues.startDate || '');
+    this.validateDate('expireDateMsg', this.props.inputValues.toDate || '');
+  }
+
+  // event handlers
+  onChangeState = (key, value) => {
+    if (key === 'assignmentName') this.validateAssignmentName(value);
+    if (key === 'startDate' || key === 'expireDate') this.validateDate(`${key}Msg`, value);
+    this.setState({ [key]: value });
+  }
+
+  onSubmit = () => {
+    const { assignmentName, startDate, expireDate } = this.state;
+
+    this.props.actions.onSubmit({ assignmentName, startDate, expireDate });
+  }
+
+  // helpers
+  validateAssignmentName = (assignmentName) => {
+    if (!assignmentName || assignmentName == '') {
+      this.setState({ nameValidationMsg: Ember.String.loc('_assignment_name_not_set') });
+    } else if (assignmentName.length > 100) {
+      this.setState({ nameValidationMsg: Ember.String.loc('_assignment_name_over_100_chars') });
+    } else {
+      this.setState({ nameValidationMsg: '' });
+    }
+  }
+
+  validateDate = (stateKey, value) => {
+    if (!value.length) {
+      this.setState({ [stateKey]: Ember.String.loc('_date_not_set_text') });
+    } else {
+      this.setState({ [stateKey]: '' });
+    }
+  }
+
+  formatStateForComponents = () => {
+    const assignmentDetailsState = {
+      assignmentName: this.state.assignmentName,
+      startDate: this.state.startDate,
+      expireDate: this.state.expireDate,
+    };
+
+    return {
+      assignmentDetailsState,
+    };
+  }
+
+  // render
+  render() {
+    const { nameValidationMsg, startDateMsg, expireDateMsg } = this.state;
+    const { strings, actions, data } = this.props;
+    const { assignmentDetailsState } = this.formatStateForComponents();
+    const showSubmitBtn =
+      !nameValidationMsg.length && !startDateMsg.length && !expireDateMsg.length;
```

Comment:
This was a little confusing. Does that mean we only show the submit button when the fields are empty?  I thought it would be the reverse.

Reply:
The `showSubmitBtn` is `true` if all validationMsg is empty.